client/navigation/utils.ts

```diff
@@ -159,7 +177,7 @@ export const defaultCategories = {
  * @param {Array} menuItems Array of menu items.
  * @return {Array} Sorted menu items.
  */
-export const sortMenuItems = ( menuItems ) => {
+export const sortMenuItems = ( menuItems: Array< Item > ): Array< Item > => {
```

Comment:
Not sure if the linter forced you to use `Array< Item >` but another way is to use `Item[]` to represent an array of Items. I have a slight preference to the latter, but I guess both would work.